### PR TITLE
Add conversions for exponential, subtract, multiply and divide ops for Stablehlo

### LINKED ISSUE #509: [MNIST] Implement conversion of stablehlo.exponential OP to TTIR dialect
`stablehlo.exponential` OP specification can be found [here](https://openxla.org/stablehlo/spec#exponential).

Seems like we need to add support in TTIR for ExponentialOP. It can be lowered into [`ttnn.exp`](https://docs.tenstorrent.com/ttnn/latest/ttnn/ttnn/exp.html).

### LINKED ISSUE #511: [MNIST] Implement conversion of stablehlo.divide OP to TTIR dialect
`stablehlo.divide` OP specification can be found [here](https://openxla.org/stablehlo/spec#divide).

This can be converted into `TTIR.DivOp` directly.

### LINKED ISSUE #547: [Ops] Implement exponential op

### LINKED ISSUE #551: [MNIST] Implement conversion of stablehlo.subtract and stablehlo.multiply OP to TTIR dialect
Stabehlo OP specification can be found [here](https://openxla.org/stablehlo/spec)

This can be converted into `TTIR.MultiplyOp` and `TTIR.SubtractOp` directly.

### PR DESCRIPTION
**Pull request by uazizTT**
Fixes #509 
Fixes #511 
Fixes #551 
Fixes #547

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]

File: test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_SubtractOp.mlir [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// RUN: ttmlir-opt --convert-stablehlo-to-ttir %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module @jit_eltwise_subtract attributes {} {
+  func.func public @test_subtract(%arg0: tensor<13x21x3xf32>, %arg1: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
+    %0 = stablehlo.subtract %arg0, %arg1 : tensor<13x21x3xf32>
+    // CHECK: %[[C:.*]] = tensor.empty[[C:.*]]
+    // CHECK: %[[C:.*]] = "ttir.subtract"[[C:.*]]
+    return %0 : tensor<13x21x3xf32>
+  }
+}
```
Comment (mrakitaTT) [inline test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_SubtractOp.mlir:11]: Great work Usman! Can you please also add test for stablehlo.divide op conversion?

File: lib/Target/TTNN/TTNNToFlatbuffer.cpp [hunk 2]
```diff
@@ -258,6 +260,9 @@ emitTTNNOperation(FlatbufferObjectCache &cache, Operation *op,
   if (auto sqrtOp = dyn_cast<SqrtOp>(op); sqrtOp) {
     return createOperation(cache, createEltwiseOp(cache, sqrtOp), debugString);
   }
+  if (auto exponentialOp = dyn_cast<ExponentialOp>(op); exponentialOp) {
+    return createOperation(cache, createEltwiseOp(cache, exponentialOp), debugString);
+  }
```
Comment (mrakitaTT) [inline lib/Target/TTNN/TTNNToFlatbuffer.cpp:265]: Btw you also need to make changes in `runtime/lib/ttnn/program.cpp`, in function:
```
static void
run(::tt::target::ttnn::EltwiseOp const *op, ...
```

In GH-527 I've made a small refactoring of that function, but you can just add duplicated code as was previously done and then I will pick up those changes in my PR if you plan to merge before me.

File: test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_SubtractOp.mlir (hunk 1 above)
Comment (uazizTT) [reply test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_SubtractOp.mlir:11]: Thanks Marko! I added test for divide op as well.

File: include/ttmlir/Dialect/TTIR/IR/TTIROps.td [hunk 3]
```diff
@@ -266,6 +266,13 @@ def TTIR_SigmoidOp: TTIR_ElementwiseUnaryOp<"sigmoid"> {
     }];
 }
 
+def TTIR_ExponentialOp: TTIR_ElementwiseUnaryOp<"exponential"> {
```
Comment (svuckovicTT) [inline include/ttmlir/Dialect/TTIR/IR/TTIROps.td:269]: Please rename `ExponentialOp` into `ExpOp` across the PR. We want to maintain parity with ttnn lib (which tracks torch).

**Review by svuckovicTT — APPROVED** [review]

File: test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_ExponentialOp.mlir [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+// RUN: ttmlir-opt --convert-stablehlo-to-ttir %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module @jit_eltwise_exponential attributes {} {
+  func.func public @test_exponential(%arg0: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
+    %0 = stablehlo.exponential %arg0 : tensor<13x21x3xf32>
+    // CHECK: %[[C:.*]] = tensor.empty[[C:.*]]
+    // CHECK: %[[C:.*]] = "ttir.exponential"[[C:.*]]
+    return %0 : tensor<13x21x3xf32>
+  }
+}
```
Comment (mrakitaTT) [inline test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_ExponentialOp.mlir:11]: For new OPs you also need to add a test for TTIR->TTNN conversion in `test/ttmlir/Dialect/TTNN/` folder. Sorry I missed this yesterday.

**Review by mrakitaTT — APPROVED** [review]

File: test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_ExponentialOp.mlir (hunk 4 above)
Comment (uazizTT) [reply test/ttmlir/Dialect/TTIR/stablehlo_to_ttir_ExponentialOp.mlir]: Ok thanks, added ExpOp test for TTNN.

**Pull request merged**